src/lib.rs

```diff
@@ -1,17 +1,55 @@
-#![cfg(any(target_os = "macos", target_os = "ios"))]
-#![allow(clippy::missing_safety_doc, clippy::let_unit_value)]
+#![cfg(target_vendor = "apple")]
+#![allow(clippy::missing_safety_doc)]
+#![cfg_attr(docsrs, feature(doc_auto_cfg, doc_cfg_hide), doc(cfg_hide(doc)))]
+#![deny(unsafe_op_in_unsafe_fn)]
 
-#[macro_use]
-extern crate objc;
-
-use objc::runtime::Object;
+use objc2::rc::Retained;
+use objc2_quartz_core::CAMetalLayer;
+use std::ffi::c_void;
 
+#[cfg(any(target_os = "macos", doc))]
 pub mod appkit;
+
+#[cfg(any(not(target_os = "macos"), doc))]
 pub mod uikit;
 
-pub type CAMetalLayer = *mut Object;
+/// A wrapper around [`CAMetalLayer`].
+pub struct Layer {
+    layer: Retained<CAMetalLayer>,
+    pre_existing: bool,
+}
+
+impl Layer {
+    /// Get a pointer to the underlying [`CAMetalLayer`]. The pointer is valid
+    /// for at least as long as the [`Layer`] is valid, but can be extended by
+    /// retaining it.
+    ///
+    ///
+    /// # Example
+    ///
+    /// ```no_run
+    /// use objc2::rc::Retained;
+    /// use objc2_quartz_core::CAMetalLayer;
+    /// use raw_window_metal::Layer;
+    ///
+    /// let layer: Layer;
+    /// # layer = unimplemented!();
+    ///
+    /// let layer: *mut CAMetalLayer = layer.as_ptr().cast();
+    /// // SAFETY: The pointer is a valid `CAMetalLayer`.
+    /// let layer = unsafe { Retained::retain(layer).unwrap() };
```

Comment:
We don't want to give access to `self.layer` directly because that publicly exposes our dependency on `objc2`.

> Or have the default example document that by keeping the `Layer` alive, they do not need to retain the pointer?

We _could_ do that, I just think it'd be easier for the user to just store `Retained<CAMetalLayer>`, rather than having to call `unsafe { &*layer.as_ptr().cast::<CAMetalLayer>() }` every time they want to use it.

Reply:
I won't be opposed to exposing the dependency unless you think there's going to be a lot of shift in crates, breaking dependency versions etc that make it hard to use `objc2` types here?